pkgs/development/python-modules/deepsearch-glm/default.nix

```
@@ -0,0 +1,104 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  cmake,
+  cxxopts,
+  deepsearch-toolkit,
+  docling-core,
+  fasttext,
+  fmt,
+  loguru,
+  matplotlib,
+  nlohmann_json,
+  pandas,
+  pcre2,
+  pkg-config,
+  poetry-core,
+  pybind11,
+  python-dotenv,
+  requests,
+  rich,
+  sentencepiece,
+  setuptools,
+  tabulate,
+  tqdm,
+  utf8cpp,
+  zlib,
+}:
+
+buildPythonPackage rec {
+  pname = "deepsearch-glm";
+  version = "1.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "DS4SD";
+    repo = "deepsearch-glm";
+    tag = "v${version}";
+    hash = "sha256-3sJNkrx0tTm6RMYAwV8Aha7x8dZjf4tGdds8OScRff8=";
+  };
+
+  dontUseCmakeConfigure = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  build-system = [ poetry-core ];
```

```suggestion
  build-system = [
    poetry-core
    pybind11
  ];
```
https://github.com/DS4SD/deepsearch-glm/blob/ad2f46b63499b2ce7ad694839f534cc913c3d7ad/pyproject.toml#L137-L138